scripts/release.sh

```diff
@@ -2,24 +2,79 @@
 
 VERSION=$1
 
+if [ -z "$VERSION" ]; then
+    echo "Error: Version argument is required"
+    echo "Usage: $0 <version>"
+    exit 1
+fi
+
 sed -i.bak "s/## \[Unreleased\]/## \[Unreleased\]\n\n## \[${VERSION}\] - $(TZ=Europe/Krakow date '+%Y-%m-%d')/" CHANGELOG.md
 rm CHANGELOG.md.bak 2> /dev/null
 
 sed -i.bak "/\[workspace.package\]/,/version =/ s/version = \".*/version = \"${VERSION}\"/" Cargo.toml
 rm Cargo.toml.bak 2> /dev/null
 
-sed -i.bak "/\[package\]/,/version =/ s/version = \".*/version = \"${VERSION}\"/" sncast_std/Scarb.toml
-rm sncast_std/Scarb.toml.bak 2> /dev/null
-
-scarb --manifest-path sncast_std/Scarb.toml build
-
-sed -i.bak "/\[package\]/,/version =/ s/version = \".*/version = \"${VERSION}\"/" snforge_std/Scarb.toml
-rm snforge_std/Scarb.toml.bak 2> /dev/null
+declare -a scarb_files=(
+    "sncast_std/Scarb.toml"
+    "snforge_std/Scarb.toml"
+    "crates/snforge-scarb-plugin/Scarb.toml"
+)
 
-sed -i.bak "/\[package\]/,/version =/ s/version = \".*/version = \"${VERSION}\"/" crates/snforge-scarb-plugin/Scarb.toml
-rm crates/snforge-scarb-plugin/Scarb.toml.bak 2> /dev/null
+for file in "${scarb_files[@]}"; do
+    sed -i.bak "/\[package\]/,/version =/ s/version = \".*/version = \"${VERSION}\"/" "$file"
+    rm "${file}.bak" 2> /dev/null
+done
 
+scarb --manifest-path sncast_std/Scarb.toml build
 scarb --manifest-path snforge_std/Scarb.toml build
 
 cargo update -p forge
 cargo update -p sncast
+
+declare -a doc_files=(
+    "./docs/src/getting-started/first-steps.md"
+    "./docs/src/starknet/script.md"
+    "./docs/src/appendix/scarb-toml.md"
+    "./docs/src/appendix/cheatcodes.md"
+    "./docs/src/appendix/snforge-library.md"
+    "./docs/src/testing/contracts.md"
+    "./docs/src/testing/using-cheatcodes.md"
+)
+
+for file in "${doc_files[@]}"; do
+    # Use different sed patterns based on the file
+    case "$file" in
+        "./docs/src/appendix/scarb-toml.md")
```

Comment:
This could use some improvement. Right now you hold a list of files in `doc_files`, and then iterate over it to check again if specific file is there. It's suboptimal.

You should rather iterate over the list and look for specififc string to replace (it seems there is a pattern in this, the string always looks like `snforge_std = { git = "https://github.com/foundry-rs/starknet-foundry.git", tag = "<<version>>" }` or `snforge_std = <<version>>`, the same with `sncast_std`). 

That way you get rid of unnecessary `case` statement, and get rid of over-complicated regex in favor of a simpler one. Sounds good?

Reply:
@THenry14 yes it does, it sounds better if it iterates over the list and look for specific string. I'm on it now. Will send an updated pr soon